provisioning/device/src/interfaces.ts

```diff
@@ -67,10 +62,18 @@ export interface RegistrationResult {
 /**
  * @private
  */
-export interface X509ProvisioningTransport {
-  setTransportOptions(options: ProvisioningTransportOptions): void;
+export interface PollingTransportHandlers {
+  registrationRequest(request: RegistrationRequest, callback: (err?: Error, body?: any, result?: any, pollingInterval?: number) => void): void;
+  queryOperationStatus(request: RegistrationRequest, operationId: string, callback: (err?: Error, body?: any, result?: any, pollingInterval?: number) => void): void;
   cancel(callback: (err?: Error) => void): void;
-  registerX509(request: RegistrationRequest, auth: X509, callback: (err?: Error, registrationResult?: RegistrationResult, body?: any, result?: any) => void): void;
```

Comment:
huh - where are these going? this is still the public interface for the transport, the register function in the PollingStateMachine is still hidden under it right?

Reply:
never mind - RegistrationClient is the public interface. my bad.

---
In reply to: [155605404](https://github.com/Azure/azure-iot-sdk-node/pull/176#discussion_r155605404) [](ancestors = 155605404)